### PR TITLE
PCHR-2267: Appears unformatted due date in document list

### PR DESCRIPTION
**Pull request by bmnepali**
## Overview
When a document is updated with new status, the old document is replaced with the just updated document, whose dates are not formatted to display date only, resulting to the display the same date that is fetched form server, and displayed in due date text in document list with "00:00:00".

screen shot 2017-07-13 at 10 12 06 pm

The issue is fixed by removing the code that is called to update the document by the new just updated document, and `status_id` was updated to reflect the change in document status.

## Before
Due date is replaced by the un formatted date from server:
staging-documents-due-date-bug 1

## After
Due data is not updated, stays the same and on document edit modal correct dates are displayed.
final-min

## Technical Details
Make some modifications in file "uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js"

1. Refactored code to update the list and document on `$rootScope.$broadcast('documentFormSuccess')`
```js
$scope.$on('documentFormSuccess', function (e, newData, ondData) {
  if (angular.equals({}, oldData)) {
     addRemoveDocument($scope.list, newData, oldData);
  } else {
     angular.extend(oldData, newData);
  }
});
```
2. Do not call to update the document once the document status is updated.
```js
$scope.changeStatus = function (document, statusId) {
  ...
  DocumentService.save({
    id: document.id,
    status_id: statusId
  }).then(function (results) {
     document.id = results.id;
     document.status_id = results.status_id;
     $rootScope.$broadcast('documentFormSuccess', results, document);
     ... 
});
```
## Comments
- Added test cases for necessary modification to change the document status once the document status is updated.
---

- [x] Tests Pass

### CODE REVIEW

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
-/* globals angular */
 /* eslint-env amd, no-mixed-operators:false */
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js]:
```js
no-mixed-operators:false
```
what's this?

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/document-list_test.js [hunk 2]
```diff
@@ -40,6 +46,41 @@ define([
       });
     });
 
+    describe('$scope.changeStatus()', function () {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/document-list_test.js]:
```js
describe('changeStatus()', function () {
```

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/document-list_test.js [hunk 3]
```diff
@@ -40,6 +46,41 @@ define([
...
+      beforeEach(function () {
+        initController();
+        $scope.document = mockDocument;
+      });
+
+      afterEach(function () {
+        $rootScope.$apply();
+      });
+
+      describe('status is empty', function () {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/document-list_test.js]: "when the status is empty"

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/document-list_test.js [hunk 4]
```diff
@@ -40,6 +46,41 @@ define([
...
+        beforeEach(function () {
+          $scope.changeStatus($scope.document, null);
+        });
+
+        it('does not call Document service to save the status of the document', function () {
+          expect(DocumentService.save).not.toHaveBeenCalled();
+        });
+      });
+
+      describe('status is not empty', function () {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/document-list_test.js]: "when the status is not empty"

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/document-list_test.js [hunk 5]
```diff
@@ -40,6 +46,41 @@ define([
...
+      afterEach(function () {
+        $rootScope.$apply();
+      });
+
+      describe('status is empty', function () {
+        beforeEach(function () {
+          $scope.changeStatus($scope.document, null);
+        });
+
+        it('does not call Document service to save the status of the document', function () {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/document-list_test.js]: try to keep the description "conversational" unless you absolutely need to

"it does not update the document"

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/document-list_test.js [hunk 6]
```diff
@@ -40,6 +46,41 @@ define([
...
+          expect(DocumentService.save).not.toHaveBeenCalled();
+        });
+      });
+
+      describe('status is not empty', function () {
+        beforeEach(function () {
+          $scope.changeStatus($scope.document, '4');
+        });
+
+        it('calls Document service to update status of the document', function () {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/document-list_test.js]: "it updates the document"

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js (hunk 1 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js]: Refer to http://eslint.org/docs/rules/no-mixed-operators. This is to avid pre-commit earnings about mixing operators like "||" and "&&" during comparison. removes it by refactoring the code.

**Pull request merged**